wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,163 @@
+  and other actors want to use a new eviction API to gracefully remove pods. This would enable new
+  migration strategies that prefer to surge (upscale) pods first rather than downscale them. It
+  would also allow other users/components to monitor pods that are gracefully removed/terminated
+  and provide better behaviour in terms of de/scheduling, scaling and availability.
+- As a cluster admin, I want to be able to perform arbitrary actions after the node drain is
+  complete, such as resetting GPU drivers, resetting NICs, performing software updates or shutting
+  down the machine.
+- As an end user, I would like more alternatives to blue-green upgrades, especially with special
+  hardware accelerators; it's far too expensive. I would like to choose a strategy on how to
+  coordinate the node drain and the upgrade to achieve better cost-effectiveness.
+- As a cloud provider, I need to perform regular maintenance on the hardware in my fleet. Enhancing
+  Kubernetes to help CSPs safely remove hardware will reduce operational costs.
+- Modelling the cost of doing accelerator maintenance in today's world can be massive. And since
+  hardware accelerators tend to need more love and care, having software support to coordinate
+  maintenance will reduce operational costs.
+- As a cluster admin, I would like to use a mixture of on-demand and temporary spot instances in my
+  clusters to reduce cloud expenditure. Having more reliable lifecycle and drain mechanisms for
+  nodes will improve cluster stability in scenarios where instances may be terminated by the cloud
+  provider due to cost-related thresholds.
+- As a user, I want to prevent any disruption to my pet or expensive workloads (VMs, ML with
+  accelerators) and either prevent termination altogether or have a reliable migration path. 
+  Features like `terminationGracePeriodSeconds` are not sufficient as the termination/migration can
+  take hours if not days.
+- As a user, I want my application to finish all network and storage operations before terminating a
+  pod. This includes closing pod connections, removing pods from endpoints, writing cached writes
+  to the underlying storage and completing storage cleanup routines.
+
+## Deliverables
+
+The WG will coordinate requirement gathering and design, eventually leading to
+KEP(s)s and code associated with the ideas.
+
+Area we expect to explore:
+
+- An API to express node drain/maintenance.
+  Currently tracked in https://github.com/kubernetes/enhancements/issues/4212.
+- An API to solve the problems wrt the API-initiated Eviction API and PDBs.
+  Currently tracked in https://github.com/kubernetes/enhancements/issues/4563.
+- An API/mechanism to gracefully terminate pods during a node shutdown.
+  Graceful node shutdown feature tracked in https://github.com/kubernetes/enhancements/issues/2000.
+- An API to deschedule pods that use DRA devices.
```

Comment:
Are you saying there will be separate API for descheduling any Pod and a Pod with DRA device? Why both can't just use `/evict`?

Reply:
This just references an existing feature without specifying the implementation details here.